src/model/entities/action.ts

```diff
@@ -0,0 +1,28 @@
+import { ActionType } from '../enums/action-type'
+import { PartInterface } from './part'
+import { Piece, PieceInterface } from './piece'
+
+export interface Action {
+  id: string
+  name: string
+  type: ActionType
+  data: unknown
+}
+
+export interface PartAction extends Action {
+  type: ActionType.INSERT_PART_AS_ON_AIR | ActionType.INSERT_PART_AS_NEXT
```

Comment:
I don't know if I am a fan of this or not.
In my head, there would ideally be a `PieceActionType` and a `PartActionType`. Then `Action` could be the union of those or `Action.type` could be something more generic like `string`.

Reply:
Then you are introducing two "values" to tell what kind of action is and when you need to figure out which kind of Action you are dealing with you need to do something like:
```
switch(action.type) {
  case ActionType.PIECE: {
    switch(action.subType) {
       case ActionSubType.INSER_AS_ON_AIR: {
         ...
       }
       case ActionSubType.INSER_AS_NEXT: {
         ...
       }
    }
  }
  case: ActionType.PART: {
    switch(action.subType): {
      case ActionSubType.INSER_AS_ON_AIR: {
        ...
      }
      case ActionSubType.INSER_AS_NEXT: {
        ...
      }
    }
  }
}
```
I would very much like us to avoid that...

Reply:
I might have formulated it badly, but you would not need the subtype attribute. I was thinking something like:

```typescript
type ActionType = PieceActionType | PartActionType

enum PieceActionType {
  INSERT_PIECE_AS_ON_AIR = 'INSERT_PIECE_AS_ON_AIR',
  INSERT_PIECE_AS_NEXT = 'INSERT_PIECE_AS_NEXT',
  // ...
}

enum PartActionType {
  INSERT_PART_AS_ON_AIR = 'INSERT_PART_AS_ON_AIR',
  INSERT_PART_AS_NEXT = 'INSERT_PART_AS_NEXT',
  // ...
}

interface Action {
  type: ActionType
  // ...
}

interface PartAction extends Action {
  type: PartActionType
  // ...
}

interface PieceAction extends Action {
  type: PieceActionType
  // ...
}

```

Reply:
I have updated to your suggestion.